metrics_sdk/lib/opentelemetry/sdk/metrics/aggregation/sum.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module SDK
+    module Metrics
+      module Aggregation
+        # Contains the implementation of the Sum aggregation
+        # https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#sum-aggregation
+        class Sum
+          def initialize
+            @aggregation_temporality = :delta
+
+            @data_points = {}
+            @mutex = Mutex.new
```

Comment:
I didn't see where this was being used, but maybe I'm missing something. I mean, I think the idea of a mutex around updates might be desirable (see @fbogsany's comments on the histogram aggregation), don't get me wrong. 😄

Reply:
Good catch, this is not being used. It is remnants of a previous iteration.

Reply:
Removed e42584135ba9e15a9bafeb18a58a2f266f6e71f1